docs/kubefirst/github/gitops.md

```diff
@@ -0,0 +1,78 @@
+# Kubefirst GitOps CI/CD
+
+## Say Goodbye to Deployment Scripts with GitOps
+
+With our GitOps patterns, the days of scripting `kubectl apply` and `helm install` operations are over. Your pipeline's job to deploy code is to simply to set the version of the app you want in the `main` branch of the `gitops` repository.
+
+## The Benefits of GitOps
+
+The benefits of defining your desired state in a declarative way in git is enormous.
+
+- you get an audit log of every system change and who made it and who approved it
+- securing the main branch of the gitops repo makes for very easy security and approval controls for system changes
+- easy to understand rollbacks (just revert the git commit that caused the problem and you'll be on the last version)
+- declarative state in git significantly reduces the complexity of disaster recovery
+- creating new environments is as straight-forward as duplicating directory structures
+- having everything declared in the gitops repo means everyone can agree to the source of truth for every version of every app
+
+## See How Rollbacks Work
+<div class="video-wrapper">
+  <iframe width="1280" height="500" src="https://www.youtube.com/embed/iqZA9Eycvgo" frameborder="0" allowfullscreen></iframe>
+</div>
+
+## Pipelines
+
+![](../../img/kubefirst/gitops/gitops-cicd.png)
+
+Our CI pipelines originate in your privately hosted GitHub Actions Runner with a workflow being submitted to Argo Workflows. The workflows will publish your container with a pre-release helm chart.
+
+The pipelines then continue with promotion of the release through the preprod environments.
+
+Once the release is ready, the release chart is published and delivered to production, and the application chart's version will be prepared for the next release.
+
+## GitHub Integration with Argo Workflows
+
+![](../../img/todo.jpeg)
+
+`todo: need new image for github integration`
+
+![](../../img/kubefirst/gitops/gitlab-workflows-integration.png)
+
+The integration that we've established between GitHub and Argo Workflows will surface your powerful Argo Workflows engine directly in GitHub Actions, so that your developers have all of the workflow execution information available directly associated with their commit in their application repository.
```

Comment:
```suggestion
The integration that we've established between GitHub and Argo Workflows will surface your powerful Argo Workflows engine directly in GitHub Actions, so that your developers have all of the workflow execution information available, directly associated with their commit in their application repository.
```